pkg/apis/batch/v1/types.go

```diff
@@ -48,6 +48,29 @@ type JobList struct {
 	Items []Job `json:"items"`
 }
 
+// JobTemplate describes a template for creating copies of a predefined pod.
+type JobTemplate struct {
+	unversioned.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+	v1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Template defines jobs that will be created from this template
+	// http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+	Template JobTemplateSpec `json:"template,omitempty"`
+}
+
+// JobTemplateSpec describes the data a Job should have when created from a template
+type JobTemplateSpec struct {
+	// Standard object's metadata of the jobs created from this template.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+	v1.ObjectMeta `json:"metadata,omitempty"`
```

Comment:
This problem exists in all existing controllers also, but we should really create a stripped-down ObjectMeta to use in templates that only contains the relevant fields. The other fields are a liability to evolution, since the fields will actually be decoded and persisted, and make our documentation misleading and our schema validation incorrect.